pkg/smokescreen/smokescreen.go

```diff
@@ -255,7 +255,11 @@ func dialContext(ctx context.Context, network, addr string) (net.Conn, error) {
 		return nil, err
 	}
 	sctx.cfg.StatsdClient.Incr("cn.atpt.success.total", []string{}, 1)
-
+	
+	if sctx.cfg.ReverseProxyProtocol {
+		conn = proxyproto.NewConn(conn, sctx.cfg.ConnectTimeout)	
```

Comment:
Looking at the library, I don't think this will do what we want. `NewConn` creates a new connection that listens for an incoming proxy protocol header (i.e. from the host we're connecting to), rather than sending a proxy protocol header onwards.